### DISCUSSION TITLE
Save content of button response not its index

### DISCUSSION THREAD

**Discussion by valentinavellani** [jspsych/jsPsych#1687] [Q&A] [answered]
Hi,
in this code I show 3 buttons in a random order.
I would like to save the information about which button has been chosen, but I cannot use the index as the order changes at every trial.
Ideally, I would like to have a number associated to each choice and to store this number,

```js
var use_info = {
    type: 'html-button-response',
    stimulus: '<span style="font-size:36px; ">What would you like to do?</span>',
    choices: function(data) {
         if (show==1){
            
            list_options= gorilla.shuffle (['<span style="font-size:30px; ">gather 5 stocks</span>','<span style="font-size:30px; ">nothing</span>', '<span style="font-size:30px; ">give away 5 stocks</span>'])
         return list_options
             
         } else if (show==0){
         
            list_options= gorilla.shuffle(['<span style="font-size:30px; ">nothing</span>', '<span style="font-size:30px; ">nothing</span>' , '<span style="font-size:30px; ">nothing</span>'])
           return list_options
             
         }
    },
    on_finish: function(data) {
            decision = jsPsych.data.getLastTrialData().values()[0].button_pressed
    }
};
timeline.push(use_info);
```

Thank you very much for your help!
Best
Valentina

**Comment by becky-gilbert** [comment, marked as answer]
Hi Valentina,

Great question. If you're using jsPsych v6.3.0, there's an easy way to do this. You can save the randomized button choice labels to the trial data using the new [`save_trial_parameters`](https://www.jspsych.org/overview/plugins/#the-save_trial_parameters-parameter) parameter that's available in all plugins. And in the `on_finish` function, you could also add a property to the trial data that contains the label of the button that was pressed. In the jsPsych examples folder, there's an example file called "save_trial_parameters.HTML" - the 3rd example trial in that file shows exactly how to do this.

If you're using a version of jsPsych <6.3.0, then the `save_trial_parameters` option doesn't exist. So another way to do this is to make the `list_options` array into a global variable, so that you can access the current value of that variable in the trial's `on_finish` function. That way you can use the `button_pressed` index to figure out which button was pressed.
```js
// make list_options a global variable
var list_options;
var use_info = {
    type: 'html-button-response',
    stimulus: '<span style="font-size:36px; ">What would you like to do?</span>',
    choices: function(data) {
         if (show==1){
            list_options= gorilla.shuffle (['<span style="font-size:30px; ">gather 5 stocks</span>','<span style="font-size:30px; ">nothing</span>', '<span style="font-size:30px; ">give away 5 stocks</span>'])
             return list_options  
         } else if (show==0){
             list_options= gorilla.shuffle(['<span style="font-size:30px; ">nothing</span>', '<span style="font-size:30px; ">nothing</span>' , '<span style="font-size:30px; ">nothing</span>'])
             return list_options
         }
    },
    on_finish: function(data) {
         // find the label of the button that was pressed by indexing the current list_options array with the button_pressed index
         data.decision = list_options[data.button_pressed];
    }
};
timeline.push(use_info);
```
Note that, in the `on_finish` function, you don't need to use `jsPsych.data.getLastTrialData().values()[0]` to get the trial data, because the trial data is passed into the `on_finish` function as the `data` object. So you can just do `data.button_pressed`. Also, if you want to add information to the trial data, then you need to add it to the `data` object, e.g. `data.decision = ...`. 

I hope this helps!

**Comment by valentinavellani** [comment]
Hi Becky,
thank you very much for your answer! I am using a version of jsPsych <6.3.0 so I went for the second option.

Thanks again!
Best
Valentina
